### DISCUSSION TITLE
How to load and install duckdb extension from R

### DISCUSSION THREAD

**Discussion by handevries** [duckdb/duckdb#6989] [Q&A] [answered]
I'm trying to install and load the fts extension using the DuckDB R API. It seems the install step works fine, but loading fts causes R to crash. With a fresh installation of R and the duckdb package I get:
> library(DBI)
> library(duckdb)
> con <- dbConnect(duckdb::duckdb())
> sessionInfo()
R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=Dutch_Netherlands.utf8  LC_CTYPE=Dutch_Netherlands.utf8   
[3] LC_MONETARY=Dutch_Netherlands.utf8 LC_NUMERIC=C                      
[5] LC_TIME=Dutch_Netherlands.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] duckdb_0.7.1-1 DBI_1.1.3     

loaded via a namespace (and not attached):
[1] compiler_4.2.3 tools_4.2.3   
> dbExecute(con, "INSTALL 'fts';")
[1] 0
> dbExecute(con, "LOAD 'fts';") 

"Process R exited abnormally with code 5"

------------------
The problem seems to persist with other extensions (like json). However, it works fine outside R, i.e. when using the DuckDB CLI.
Downloading the extension manually, and using the pathname instead of 'fts' didn't resolve the problem either.

**Comment by rsund** [comment, marked as answer]
MinGW toolchain appears to be the problem: [https://github.com/duckdb/duckdb/issues/6231#issuecomment-1431044140](https://github.com/duckdb/duckdb/issues/6231#issuecomment-1431044140).

Some in-tree extensions (including FTS and JSON) are/were included in the R-universe Windows binaries (i.e. extensions are available without INSTALL or LOAD), so installing the DuckDB R-package from there instead of CRAN could work as a partial workaround: [https://duckdb.r-universe.dev/duckdb](https://duckdb.r-universe.dev/duckdb).
